### PR TITLE
Adapt to https://github.com/coq/coq/pull/19530

### PR DESCRIPTION
**Pull request by proux01**
To be merged in sync with the upstream PR

### CODE REVIEW

**Comment by JasonGross** [comment]
> To be merged in sync with the upstream PR

Coq Tools should not require synchronized merges; the bug minimizer needs to work across multiple versions of Coq.  What's the issue?

File: examples/run-example-048.sh [hunk 1]
```diff
@@ -89,7 +89,7 @@ find_bug "$EXAMPLE_INPUT" "$EXAMPLE_OUTPUT" "${EXTRA_ARGS[@]}" || exit $?
 (\* -\*- mode: coq; coq-prog-args: ([^)]*) -\*- \*)
 (\* File reduced by coq-bug-minimizer from original input, then from [0-9]\+ lines to [0-9]\+ lines, then from [0-9]\+ lines to [0-9]\+ lines, then from [0-9]\+ lines to [0-9]\+ lines, then from [0-9]\+ lines to [0-9]\+ lines \*)
 (\* coqc version [^\*]*\*)
-Require \(Coq\|Stdlib\)\.Init\.Ltac\.
+Require \(Coq\|Corelib\)\.Init\.Ltac\.
```
Comment (JasonGross) [inline examples/run-example-048.sh:92]:
```suggestion
Require \(Coq\|Corelib\|Stdlib\)\.Init\.Ltac\.
```

File: examples/run-example-048.sh [hunk 2]
```diff
@@ -100,7 +100,7 @@ Global Set Universe Polymorphism\.
 Definition foo@{i} := Type@{i}\.
 
 End Foo\.
-\(Import \(Coq\|Stdlib\)\.Init\.Ltac\.
+\(Import \(Coq\|Corelib\)\.Init\.Ltac\.
```
Comment (JasonGross) [inline examples/run-example-048.sh:103]:
```suggestion
\(Import \(Coq\|Corelib\|Stdlib\)\.Init\.Ltac\.
```

**Comment by proux01** [comment]
> > To be merged in sync with the upstream PR
> 
> Coq Tools should not require synchronized merges; the bug minimizer needs to work across multiple versions of Coq. What's the issue?

Well, I guess we could have something backward compatible but here it's about backward compatibility with an old master thing that was never released, so probably not worth it.

File: coq_tools/find_bug.py [hunk 3]
```diff
@@ -2247,7 +2247,10 @@ def make_make_coqc(coqc_prog, **kwargs):
                         list(env[passing_prefix + "libnames"]) + [(coq_theories_path, "Coq")]
                     )
                     env[passing_prefix + "libnames"] = tuple(
-                        list(env[passing_prefix + "libnames"]) + [(coq_user_contrib_path, "Coq")]
```
Comment (JasonGross) [inline coq_tools/find_bug.py:2250]: Is this change actually necessary at this point?  It breaks minimization on old versions of Coq.  Maybe you meant to replace Stdlib with Coq below?

File: coq_tools/find_bug.py [hunk 4]
```diff
@@ -2247,7 +2247,10 @@ def make_make_coqc(coqc_prog, **kwargs):
...
+                        list(env[passing_prefix + "libnames"]) + [(coq_theories_path, "Corelib")]
+                    )
+                    env[passing_prefix + "libnames"] = tuple(
+                        list(env[passing_prefix + "libnames"]) + [(coq_user_contrib_path, "Stdlib")]
```
Comment (JasonGross) [inline coq_tools/find_bug.py:2253]:
```suggestion
                        list(env[passing_prefix + "libnames"]) + [(coq_user_contrib_path, "Coq")]
```

File: coq_tools/find_bug.py [hunk 5]
```diff
@@ -2246,6 +2246,9 @@ def make_make_coqc(coqc_prog, **kwargs):
                     env[passing_prefix + "libnames"] = tuple(
                         list(env[passing_prefix + "libnames"]) + [(coq_theories_path, "Coq")]
                     )
+                    env[passing_prefix + "libnames"] = tuple(
+                        list(env[passing_prefix + "libnames"]) + [(coq_theories_path, "Corelib")]
+                    )
                     env[passing_prefix + "libnames"] = tuple(
                         list(env[passing_prefix + "libnames"]) + [(coq_user_contrib_path, "Coq")]
```
Comment (JasonGross) [inline coq_tools/find_bug.py:2253]:
```suggestion
                        list(env[passing_prefix + "libnames"]) + [(coq_user_contrib_path, "Stdlib")]
```

File: examples/run-example-048.sh (hunk 2 above)
Comment (JasonGross) [reply examples/run-example-048.sh:103]: Why did you revert this suggestion?  The build fails as a result of this.

**Comment by JasonGross** [comment]
Example 048 fails because the code is minimized to something including `Require Stdlib.Init.Ltac.`

Example 055 is failing with "Error: Cannot find a physical path bound to logical path Prelude with prefix Stdlib."  Any idea what's going wrong?

**Comment by JasonGross** [comment]
> Example 055 is failing with "Error: Cannot find a physical path bound to logical path Prelude with prefix Stdlib."

I think we need to adjust
https://github.com/JasonGross/coq-tools/blob/6f29d5f2ebbd09cc47fba048d8cffb9887bcdf43/coq_tools/find_bug.py#L2361-L2364
with a version check.  Alternatively, we can move `add_coqlib_prelude_import` to be like https://github.com/JasonGross/coq-tools/blob/6f29d5f2ebbd09cc47fba048d8cffb9887bcdf43/coq_tools/coq_running_support.py#L26-L60

**Comment by Zimmi48** [comment]
> Example 055 is failing with "Error: Cannot find a physical path bound to logical path Prelude with prefix Stdlib." Any idea what's going wrong?

This can happen if you only depend on package rocq-core and not on coq-stdlib. Could that be the case?

> Example 048 fails because the code is minimized to something including `Require Stdlib.Init.Ltac.`

If coq-stdlib is in scope, this should not be considered as a failure (unless coq-stdlib is now viewed as an external library which makes the test fail).

**Comment by proux01** [comment]
example55 works locally so I guess Theo is right and this is a CI issue. This is probably waiting for the Docker image

**Comment by JasonGross** [comment]
> > Example 048 fails because the code is minimized to something including `Require Stdlib.Init.Ltac.`
> 
> If coq-stdlib is in scope, this should not be considered as a failure (unless coq-stdlib is now viewed as an external library which makes the test fail).

Right, this is just a regex that needs fixing, I've committed the fix

**Comment by erikmd** [comment]
@Zimmi48 

> Example 055 is failing with "Error: Cannot find a physical path bound to logical path Prelude with prefix Stdlib." Any idea what's going wrong?

FYI this looks like the same error we get in building `coqorg/coq:dev` in Docker-Coq, as Bignums does not find the stdlib anymore since the recent stdlib-related changes.

after a bit of investigation and discussion with @proux01, it seems https://github.com/coq/opam/blob/master/core-dev/packages/coq-stdlib/coq-stdlib.dev/opam is **broken**: after doing `opam install coq-stdlib.dev`, no .vo from coq-stdlib is installed anymore.

I tested an alternative version of this .opam file, to no avail:

```
opam-version: "2.0"
synopsis: "The Coq Proof Assistant -- Standard Library"
description: """
Coq is a formal proof management system. It provides
a formal language to write mathematical definitions, executable
algorithms and theorems together with an environment for
semi-interactive development of machine-checked proofs.

Typical applications include the certification of properties of
programming languages (e.g. the CompCert compiler certification
project, or the Bedrock verified low-level programming library), the
formalization of mathematics (e.g. the full formalization of the
Feit-Thompson theorem or homotopy type theory) and teaching.

This package includes the Coq Standard Library, that is to say, the
set of modules usually bound to the Stdlib.* namespace."""
maintainer: ["The Coq standard library development team"]
authors: ["The Coq development team, INRIA, CNRS, and contributors"]
license: "LGPL-2.1-only"
homepage: "https://coq.inria.fr/"
doc: "https://coq.github.io/doc/"
bug-reports: "https://github.com/coq/coq/issues"
depends: [
  "dune" {>= "3.8"}
  "coq-core"
  "rocq-core" {= version}
  "odoc" {with-doc}
]
depopts: ["coq-native"]
dev-repo: "git+https://github.com/coq/coq.git"
# man dune (3.16.1)
# "-p" name :=
# "--release" "--only-packages" name :=
# "--root" "." "--ignore-promoted-rules" "--no-config" "--profile" "release"
# "--always-show-command-line" "--promote-install-files"
# "--require-dune-project-file" "--ignore-lock-dir"
# "--default-target" "@install" "--only-packages" name
build: [
  ["dune" "subst"] {dev}
  [
    "dune"
    "build"
    "--root" "stdlib" "--ignore-promoted-rules" "--no-config"
    "--profile" "release" "--always-show-command-line"
    "--require-dune-project-file" "--ignore-lock-dir"
    "--default-target" "@install" "--only-packages" name
    "-j"
    jobs
    "--promote-install-files=false"
    "@install"
    "@runtest" {with-test}
    "@doc" {with-doc}
  ]
  ["dune" "install" "--root" "stdlib" "--ignore-promoted-rules" "--no-config"
   "--profile" "release" "--always-show-command-line"
   "--promote-install-files" "--require-dune-project-file" "--ignore-lock-dir"
   "--default-target" "@install" "--only-packages" name
   "--create-install-files" name]
]

url {
  src: "git+https://github.com/coq/coq.git#master"
}
```


so @proux01 should investigate further on this Monday

**Comment by proux01** [comment]
Sorry for the delay, the opam/docker issue is now fixed and CI seems happy, can this be merged?

**Comment by proux01** [comment]
Thanks for your patience!
